WordPress/src/main/java/org/wordpress/android/ui/taxonomies/TermsDataViewActivity.kt

```diff
@@ -0,0 +1,299 @@
+            onFetchMore = {
+                viewModel.onFetchMoreData()
+            },
+            onSearchQueryChange = { query ->
+                viewModel.onSearchQueryChange(query)
+            },
+            onItemClick = { item ->
+                viewModel.onItemClick(item)
+                (item.data as? AnyTermWithEditContext)?.let { term ->
+                    navController.currentBackStackEntry?.savedStateHandle?.set(
+                        key = KEY_TERM_ID,
+                        value = term.id
+                    )
+                    navController.navigate(route = TermScreen.Detail.name)
+                }
+            },
+            onFilterClick = { filter ->
+                viewModel.onFilterClick(filter)
+            },
+            onSortClick = { sort ->
+                viewModel.onSortClick(sort)
+            },
+            onSortOrderClick = { order ->
+                viewModel.onSortOrderClick(order)
+            },
+            emptyView = viewModel.emptyView,
+            modifier = modifier
+        )
+    }
+
+    @Composable
+    private fun ShowTermDetailScreen(
+        allTerms: List<AnyTermWithEditContext>,
+        term: AnyTermWithEditContext,
+        modifier: Modifier
+    ) {
+        Column(
+            modifier = modifier
+                .fillMaxSize()
+                .padding(16.dp)
+                .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(16.dp)
+        ) {
+            TermDetailsCard(allTerms, term)
+        }
+    }
+
+    @Composable
+    private fun TermDetailsCard(allTerms: List<AnyTermWithEditContext>, term: AnyTermWithEditContext) {
+        Card(
+            modifier = Modifier.fillMaxWidth(),
+            elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
+            colors = CardDefaults.cardColors(
+                containerColor = MaterialTheme.colorScheme.surface
+            )
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                verticalArrangement = Arrangement.spacedBy(12.dp)
+            ) {
+                DetailRow(
+                    label = stringResource(R.string.term_name_label),
+                    value = term.name
+                )
+
+                DetailRow(
+                    label = stringResource(R.string.term_slug_label),
+                    value = term.slug
+                )
+
+                DetailRow(
+                    label = stringResource(R.string.term_description_label),
+                    value = term.description
+                )
+
+                DetailRow(
+                    label = stringResource(R.string.term_count_label),
+                    value = term.count.toString()
+                )
+
+                term.parent?.let { parentId ->
+                    val parentName = allTerms.firstOrNull {  it.id == parentId }?.name
```

Comment:
Asking for my own education to better understand Jetpack Compose: are there downsides (performance?) to passing the entire list of items for filtering the parent name? Should we instead provide the parent or its name to `TermDetailCard`? I.e., perform the filtering higher in the Compose tree.